whylog/config/parsers.py

```diff
@@ -49,3 +48,35 @@ def create_from_intent(cls, parser_intent):
             str(uuid.uuid4()), parser_intent.regex, parser_intent.primary_key_groups,
             parser_intent.log_type_name, parser_intent.data_conversions
         )
+
+
+class ConcatedRegexParser(object):
+    def __init__(self, parser_list):
+        self._parsers = parser_list
+        self._regex = re.compile(self._create_concated_regex())
+        self._parsers_indexes = self._get_indexes_of_groups_for_parsers()
+
+    def _create_concated_regex(self):
+        return "|".join(["(" + parser.regex_str + ")" for parser in self._parsers])
+
+    def _get_indexes_of_groups_for_parsers(self):
+        indexes_dict = {}
+        free_index = 0
+        for parser in self._parsers:
+            amount_of_group = parser.regex_str.count('(') - parser.regex_str.count('\(')
+            indexes_dict[parser.name] = (free_index, amount_of_group)
+            free_index += amount_of_group + 1
+        return indexes_dict
+
+    def get_extracted_regex_params(self, line):
+        matched = self._regex.match(line)
+        if matched is None:
+            return {}
+        clues = {}
+        groups = matched.groups()
+        for name, indexes in self._parsers_indexes.items():
+            if groups[indexes[0]] is not None:
```

Comment:
clearly `indexes[0]` and `indexes[1]` have a special meaning, so they can be named, so that this code would become easier to understand.